enigma-swing/src/main/java/cuchaz/enigma/gui/util/SingleTreeSelectionModel.java

```diff
@@ -0,0 +1,11 @@
+package cuchaz.enigma.gui.util;
+
+import javax.swing.tree.DefaultTreeSelectionModel;
+import javax.swing.tree.TreeSelectionModel;
+
+public class SingleTreeSelectionModel extends DefaultTreeSelectionModel {
+
+    public SingleTreeSelectionModel() {
+        this.setSelectionMode(TreeSelectionModel.SINGLE_TREE_SELECTION);
```

Comment:
Does this have to be a new subclass? I doubt setSelectionMode would be a protected method.

If this is just there to save you from calling setSelectionMode multiple times for each usage I'd just have it be a static method somewhere that creates the model and sets that but I guess it essentially comes down to personal preference.

Reply:
I thought it would make the code clearer, and be used for other properties in the future, if needed

Reply:
Yeah, that's fair enough.